tio/bufferedreader.go

```diff
@@ -89,15 +95,16 @@ var BufferDataInvalid = bufferError("Invalid data")
 // has been reached. The latter has to be enabled by flag and will disable the
 // default behavior, which is looking for a delimiter string.
 type BufferedReader struct {
-	data       []byte
-	delimiter  []byte
-	parse      func() ([]byte, int)
-	paramMLE   int
-	growSize   int
-	end        int
-	encoding   binary.ByteOrder
-	flags      BufferedReaderFlags
-	incomplete bool
+	data         []byte
+	delimiter    []byte
+	parse        func() ([]byte, int)
+	paramMLE     int
+	growSize     int
+	end          int
+	encoding     binary.ByteOrder
+	flags        BufferedReaderFlags
+	incomplete   bool
+	delimiterStr string
```

Comment:
instead of storing the string here, store the compiled regex, so regex compiling is done once during initialisation.